crates/common/src/tree.rs

```diff
@@ -373,17 +398,19 @@ where
                 let account_key_hash = KeyHash::with::<Hasher>(hashed_id);
 
                 // Verify that the account doesn't already exist
-                if self.get(account_key_hash)?.is_ok() {
+                if matches!(self.get(account_key_hash)?, HashchainResponse::Found(_, _)) {
                     bail!(DatabaseError::NotFoundError(format!(
                         "Account already exists for ID {}",
                         id
                     )));
                 }
 
                 let service_key_hash = KeyHash::with::<Hasher>(Digest::hash(service_id.as_bytes()));
-                let service_hashchain = self.get(service_key_hash)?.map_err(|_| {
-                    anyhow!("Failed to get hashchain for service ID {}", service_id)
-                })?;
+
+                let HashchainResponse::Found(service_hashchain, _) = self.get(service_key_hash)?
+                else {
+                    bail!("Failed to get hashchain for service ID {}", service_id);
+                };
```

Comment:
Same comment about my preference for match... trying to find a good reasoning for my preference and I think the reason I find it more unreadable is because:

1. The declaration of the variable gets obscured (instead of `let service_hashchain: Hashchain = ...`, it gets nested)
2. Its a new rust syntax so I haven't seen it very often in the wild
3. It is clearer in a match that the bail is for `HashchainResponse::NotFound`, without the reader needing to look at the enum variants

But this is a non-blocking nit, if you want to keep this syntax it's fine